File: financeiro-api/src/main/java/br/com/iftm/financeiroapi/model/service/CategoryService.java
package br.com.iftm.financeiroapi.model.service;

import br.com.iftm.financeiroapi.model.domain.Category;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface CategoryService {

    Category save(Category category);

    void delete(String id);

    List<Category> findAll();

    Category findOne(Long id);

}

File: financeiro-api-kotlin/src/main/kotlin/br/com/iftm/financeiroapi/model/service/CategoryService.kt
package br.com.iftm.financeiroapi.model.service

import br.com.iftm.financeiroapi.model.domain.Category
import org.springframework.stereotype.Service

@Service
interface CategoryService {

    fun save(category: Category): Category

    fun delete(category: Category)

    fun findAll(): List<Category>

    fun findOne(id: Long): Category

}